src/util-decode-mime.c

```diff
@@ -1265,21 +1265,40 @@ static uint32_t ProcessBase64Remainder(
     return buf_consumed;
 }
 
-static inline MimeDecRetCode ProcessBase64BodyLineCopyRemainder(
+/**
+ * \brief Skip any characters outside of the base64 alphabet as RFC 2045
+ * is followed for MIME parsing. Also, update bvr_len accordingly as if the
+ * DecodeBase64 returned more than 4 Bytes in there, it is likely because
+ * of the presence of invalid characters in the encoded string.
+ *
+ * \param buf The current line
+ * \param len The length of the line
+ * \param offset The offset where to start reading the data from
+ * \param state The current parser state
+ * */
+static inline MimeDecRetCode B64SkipUnneededCharsLastBlk(
         const uint8_t *buf, const uint32_t buf_len, const uint32_t offset, MimeDecParseState *state)
 {
-    DEBUG_VALIDATE_BUG_ON(offset > buf_len);
-    if (offset > buf_len)
-        return MIME_DEC_ERR_DATA;
-
     for (uint32_t i = offset; i < buf_len; i++) {
-        if (buf[i] != ' ') {
-            DEBUG_VALIDATE_BUG_ON(state->bvr_len >= B64_BLOCK);
-            if (state->bvr_len >= B64_BLOCK)
+        if (IsBase64Alphabet(buf[i])) {
+            DEBUG_VALIDATE_BUG_ON(state->bvr_len > B64_BLOCK);
+            if (state->bvr_len > B64_BLOCK)
```

Comment:
bvr_len == 4 not caught, will overflow. Check should be `>= B64_BLOCK`

Reply:
🙇🏽‍♀️ I have been overlooking this since the very beginning and doing this off by one.. Need to investigate a bit more.